File: ios/Classes/controller/LoggingController.swift
import Flutter
import WebRTC

/// Controller of the global logging configuration.
class LoggingController {
  /// Method channel for communicating with Flutter side.
  private var channel: FlutterMethodChannel

  /// Initializes a new `LoggingController` and binds method channel handlers.
  init(messenger: FlutterBinaryMessenger) {
    let channelName = ChannelNameGenerator.name(
      name: "logging",
      id: 0
    )
    self.channel = FlutterMethodChannel(
      name: channelName,
      binaryMessenger: messenger
    )
    self.channel.setMethodCallHandler { call, result in
      self.onMethodCall(call: call, result: result)
    }
  }

  /// Handles all the supported Flutter method calls.
  func onMethodCall(call: FlutterMethodCall, result: FlutterResult) {
    let argsMap = call.arguments as? [String: Any]
    switch call.method {
    case "setLogLevel":
      let level = argsMap?["level"] as? Int
      let severity: RTCLoggingSeverity
      switch level {
      case 0:
        severity = .verbose
      case 1:
        severity = .info
      case 2:
        severity = .warning
      case 3:
        severity = .error
      default:
        severity = .warning
      }

      RTCSetMinDebugLogLevel(severity)
      result(nil)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/com/instrumentisto/medea_flutter_webrtc/controller/LoggingController.kt
package com.instrumentisto.medea_flutter_webrtc.controller

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import org.webrtc.Logging

/** Controller of the global logging configuration. */
class LoggingController(messenger: BinaryMessenger) : Controller {
  /** Channel listened for the [MethodCall]s. */
  private val chan = MethodChannel(messenger, ChannelNameGenerator.name("logging", 0))

  init {
    ControllerRegistry.register(this)
    chan.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
    when (call.method) {
      "setLogLevel" -> {
        val level: Int? = call.argument("level")
        val severity =
            when (level) {
              0 -> Logging.Severity.LS_VERBOSE
              1 -> Logging.Severity.LS_INFO
              2 -> Logging.Severity.LS_WARNING
              3 -> Logging.Severity.LS_ERROR
              else -> Logging.Severity.LS_WARNING
            }
        Logging.enableLogToDebugOutput(severity)
        result.success(null)
      }
      else -> result.notImplemented()
    }
  }

  /** Releases all the allocated resources. */
  override fun dispose() {
    ControllerRegistry.unregister(this)
    chan.setMethodCallHandler(null)
  }
}
